src/dataflow/src/server/storage_state.rs

```diff
@@ -136,6 +200,8 @@ impl<'a, A: Allocate, B: StorageCapture> ActiveStorageState<'a, A, B> {
                         self.storage_state.source_descriptions.remove(&id);
                         self.storage_state.source_uppers.remove(&id);
                         self.storage_state.reported_frontiers.remove(&id);
+                        self.storage_state.ts_histories.remove(&id);
+                        self.storage_state.ts_source_mapping.remove(&id);
```

Comment:
No strong feelings, but calling out that this removes the `tracing::debug!` calls that were here previously. Maybe that is fine, because things are less likely to be out of sync, and we are already not tracking whether we drop a thing that isn't created.

Reply:
Yeah this is the rational for their removal. The binding state tear down now has the same expectations and complexity as the other ones in this function that didn't need to be explicit about checking the invariant so I removed them

I addressed all the other comments and enabled automerge